wpimath/src/main/java/edu/wpi/first/math/geometry/Rotation2d.java

```diff
@@ -153,6 +153,24 @@ public Rotation2d rotateBy(Rotation2d other) {
         m_cos * other.m_cos - m_sin * other.m_sin, m_cos * other.m_sin + m_sin * other.m_cos);
   }
 
+  /**
+   * Returns the absolute value of the Rotation2d 
+   * 
+   * @return The new absolute value Rotation2d
+   */
+  public Rotation2d abs() {
+    return new Rotation2d(Math.abs(m_value));
+  }
+
+  /**
+   * Inverts the Rotation2d, unlike `unaryMinus`, which is a flip across the x-axis, this rotates the angle 180 degrees about the origin.
```

Comment:
This is not how you invert a rotation. For a rotation A, the inverse of A is defined as the rotation that, when applied to A, gives a rotation of zero. If the heading of A is 45 degrees, this function says the inverse is 180+45=225 degrees. Applying this to A gives 45+225=270 degrees. This is not zero degrees.

The implementation of unaryMinus() is correct.

Reply:
My bad then, I wanted something that would flip the angle around, so 30 becomes -150, 0 -> 180, -150 -> 30. This is a function I've used several times to, for example, switch between the angle the robot is to the hub, and the angle the hub is to the robot